src/meta/backup_engine.cpp

```diff
@@ -215,51 +237,56 @@ void backup_engine::on_backup_reply(error_code err,
     // if backup failed, receive ERR_LOCAL_APP_FAILURE;
     // backup not completed in other cases.
     // see replica::on_cold_backup() for details.
-    int32_t partition = pid.get_partition_index();
-    if (err == dsn::ERR_OK && response.err == dsn::ERR_OK &&
-        response.progress == cold_backup_constant::PROGRESS_FINISHED) {
+    if (err != ERR_OK) {
```

Comment:
```c++
err = err == ERR_OK?  resp.err : err;

if err == ERR_LOCAL_APP_FAILURE {
   handle_replica_backup_failed
   return
}

if (err != ERR_OK) {
  retry.... 
}
```


The you just check `err`,  and don't need repeat L256

Reply:
It's better to write easy reading codes than simple codes

Reply:
Your code has been duplicated,  why the new code is not-reading?

Reply:
response is a const variable